src/components/SubPage/convert-data.ts

```diff
@@ -0,0 +1,133 @@
+import { isEmpty } from "lodash";
+import {
+    Diagram,
+    Clone,
+    DiseaseCellLineNode,
+    Sequence,
+} from "../../component-queries/types";
+import { DiagramCardProps } from "../shared/DiagramCard";
+import {
+    ClonePercentPositive,
+    UnpackedDiseaseCellLineFull,
+    UnpackedEditingDesign,
+} from "./types";
+
+export const unpackDiagrams = (diagrams?: Diagram[]): DiagramCardProps[] => {
+    if (!diagrams || diagrams.length === 0) {
+        return [];
+    }
+    return diagrams.map((diagram) => {
+        if (!diagram.image) {
+            return {
+                title: diagram.title,
+                caption: diagram.caption,
+            };
+        }
+        return {
+            title: diagram.title,
+            caption: diagram.caption,
+            image: diagram.image.childImageSharp.gatsbyImageData,
+        };
+    });
+};
+
+// subpage data
+export const unpackEditingDesignData = (editing_design?: {
+    crna_target_site?: string;
+    dna_donor_sequence?: Sequence[];
+    cas9?: string;
+    f_primer?: string;
+    r_primer?: string;
+    diagrams?: Diagram[];
+}): UnpackedEditingDesign | null => {
+    if (!editing_design) {
+        return null;
+    }
+    const diagrams = unpackDiagrams(editing_design.diagrams);
+    const data = {
+        crnaTargetSite: editing_design.crna_target_site,
+        dnaDonorSequence: editing_design.dna_donor_sequence,
+        cas9: editing_design.cas9,
+        fPrimer: editing_design.f_primer,
+        rPrimer: editing_design.r_primer,
+        diagrams: diagrams.length > 0 ? diagrams : undefined, // an empty array is still truthy
+    };
+
+    if (isEmpty(data)) {
+        return null;
+    } else {
+        return data;
+    }
+};
+
+export const getStemCellCharData = (clones: Clone[]) => {
+    const init: {
+        percentPositive: ClonePercentPositive[];
+        passingAntibodies: any[];
+        differentiation: any[];
```

Comment:
types?

Reply:
not yet, the data hasn't been entered yet, and sometimes that ends up changing the design, so I'm not going to format this data until we have examples